Analytics/models/pin_location_data.py

```diff
@@ -0,0 +1,344 @@
+        Add Tracker to db session
+        """
+        try:
+            db.session.add(self)
+            db.session.flush()
+        except IntegrityError as ite:
+            db.session.rollback()
+            logger.error("Error occurred when saving Tracker to session: id {}".format(self.id),
+                         ite.with_traceback(ite.__traceback__))
+
+    def delete(self) -> None:
+        """
+        Delete Tracker from db session
+        """
+        try:
+            db.session.delete(self)
+        except IntegrityError as ite:
+            db.session.rollback()
+            logger.error("Error occurred when deleting Tracker from session: id {}".format(self.id),
+                         ite.with_traceback(ite.__traceback__))
+
+    @staticmethod
+    def commit() -> None:
+        """ Commit session changes to the database """
+        db.session.commit()
+
+    @classmethod
+    def get_by_date_range(cls, tracker_id: str, start_date: datetime, end_date: datetime) -> [db.Model]:
+        """
+        Get LocationData for a Tracker between two dates
+        :param tracker_id: The Tracker's Id
+        :param start_date: The date of the oldest data to return
+        :param end_date: The inclusive date of the newest data to return
+        :return: A List of LocationData between the start_date and end_date
+        """
+        ts_start = datetime.strptime(start_date, '%d/%m/%y')
+        ts_end = datetime.strptime(end_date, '%d/%m/%y')
+
+        return cls.query.join(LocationData).filter(cls.id == tracker_id) \
+            .filter(and_(LocationData.timestamp >= ts_start, LocationData.timestamp <= ts_end)).all()
+
+    @classmethod
+    def get_by_tracker_id(cls, tracker_id: str) -> db.Model:
+        """
+        Get Tracker by id
+        :param tracker_id: Tracker id
+        :return: A Tracker with an id matching tracker_id
+        """
+        return cls.query.filter_by(id=tracker_id).first()
+
+
+class LocationData(db.Model):
+    """
+    Create Database Model to store location data and implement CRUD operations for location_data database table.
+    """
+    __tablename__ = 'location_data'
+
+    id = db.Column(db.Integer, primary_key=True)
+    timestamp = db.Column(db.DateTime)
+    latitude = db.Column(db.Float)
+    longitude = db.Column(db.Float)
+    speed = db.Column(db.Float, nullable=True)
+    heading = db.Column(db.Float, nullable=True)
+    elevation = db.Column(db.Float)
+    sat_cnt = db.Column(db.Integer)
+    fix_quality = db.Column(db.Integer)
+    signal_quality = db.Column(db.Integer)
+    battery = db.Column(db.Float, nullable=True)
+    charger = db.Column(db.Boolean, nullable=True)
+    value = db.Column(db.JSON, default="{}", nullable=True)
+
+    tracker_id = db.Column(db.Text, db.ForeignKey('tracker.id'))
+    tracker = db.relationship("Tracker", back_populates="loc_data")
+
+    def __init__(self, tracker_id: str, timestamp: Union[str, datetime], longitude: float, latitude: float,
```

Comment:
The variable timestamp is badly handled. It a reserved name. The data it expects isn't a timestamp. There is no description of how the data is expected since it can be both a datetime object or a string object.
Something like `event_time`, `measurement_date` or something like that would be better

Reply:
If it is done on there side or the timestamp variable is engraved deep into the code, it is ok to leave it but keep it in mind for next time. Still update the docs on what is expected.

Reply:
+ Refactored varable 'timestamp' changed variable name to 'measurement_date'